core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -40,13 +42,25 @@ class AdminClient(val time: Time,
                   val client: ConsumerNetworkClient,
                   val bootstrapBrokers: List[Node]) extends Logging {
 
+  @volatile var running: Boolean = true
+
+  val networkThread = new KafkaThread("admin-client-network-thread", new Runnable {
+    override def run() {
+      while (running) {
+        client.poll(Long.MaxValue)
+        // Allow user thread to get the lock on the ConsumerNetworkClient to enqueue requests.
+        this.synchronized()
```

Comment:
The `synchronized` has been removed in the patch after rebasing the patch onto the trunk. KAFKA-4820 has been committed to the trunk.